src/general/nqueens.rs

```diff
@@ -0,0 +1,153 @@
+#[allow(unused_imports)]
+use std::env::args;
+
+#[allow(dead_code)]
+#[cfg(not(test))]
+fn main() {
+    let mut board_width = 0;
+
+    for arg in args() {
+        board_width = match arg.parse() {
+            Ok(x) => x,
+            _ => 0,
+        };
+
+        if board_width != 0 {
+            break;
+        }
+    }
+
+    if board_width == 0 || board_width < 2 {
+        println!(
+            "Running algorithm with 8 as a default. Specify an alternative Chess board size for \
+             N-Queens as a command line argument.\n"
+        );
+        board_width = 8;
+    }
+
+    let board = match nqueens(board_width) {
+        Ok(success) => success,
+        Err(err) => panic!("{}", err),
+    };
+
+    println!("N-Queens {} by {} board result:", board_width, board_width);
+    print_board(&board);
+}
+
+pub fn nqueens(board_width: i64) -> Result<Vec<i64>, &'static str> {
+    let mut board_rows = vec![0; board_width as usize];
+    let mut conflict;
+    let mut current_row = 0;
+
+    loop
+    //Process by row
+    {
+        conflict = false;
+        for review_index in 0..current_row
+        //Column review
+        {
+            //Look for any conflict.
+            let left = board_rows[review_index] - (current_row as i64 - review_index as i64);
+            let right = board_rows[review_index] + (current_row as i64 - review_index as i64);
```

Comment:
Comments might help here

Reply:
You want a basic explanation what the variables represent?

Reply:
Yes

Reply:
I expanded the comments in the source.